Assets/BossRoom/Scripts/Shared/Net/ConnectionManagement/SessionManager.cs

```diff
@@ -0,0 +1,277 @@
+using System;
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    public interface ISessionPlayerData
+    {
+        bool IsConnected { get; set; }
+        ulong ClientID { get; set; }
+        void Reinitialize();
+    }
+
+    public class SessionManager<T> : MonoBehaviour where T : struct, ISessionPlayerData
```

Comment:
Any reason why this is a Monobehaviour? Having this as a pure C# class would make this way easier to manipulate

Reply:
It is currently placed as a script in a Gameobject, and uses the Awake(), Start() and OnDestroy() events. It also uses the "enabled" field to disable itself on clients. We could make it not a Monobehavior, and have BossRoomSessionManager have this as a field instead of inheriting from it, and refactor it to make it work, but I'm not sure if it would really make it better. I think it would be better to keep most of the logic in SessionManager so that is is easier to reuse

Reply:
Your Awake, Start and OnDestroy become useless once you port sessionManager to a self contained class.
Awake's logic becomes singleton logic, Start logic becomes constructor logic, OnDestroy logic becomes destructor logic.
"Enabled" is only useful if your method has Unity callbacks. Thing is, looking at our logic here, I'm not sure I see how this works (since ServerStartedHandler is registered in Start, which means there's no other monobehaviour callbacks which could benefit from enabling/disabling

Reply:
```suggestion
    // GUID binds a player to a session
    // Once that player connects to a host, the host associates the current ClientID to the player's session GUID.
    // If the player disconnects and reconnects to the same host, the session is preserved.
    public class SessionManager<T> : MonoBehaviour where T : struct, ISessionPlayerData
```

Reply:
Does that summarize it?

Reply:
Oh you're right, it doesn't even need to be inside a gameobject, I didn't think about that